buddyinfo.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nobuddyinfo
+// +build !windows,!netbsd
+
+package procfs
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// A BuddyInfo is the details parsed from /proc/[pid]/buddyinfo.
+// It is a accessed by node and zone.
+// The data is comprised of an array of free fragments of each size.
+// The sizes are 2^n*PAGE_SIZE, where n is the array index.
+type BuddyInfo map[string]map[string][]float64
```

Comment:
When I started, I thought I might need to do more processing where it would be good to directly address each line.  Thinking about it, this could probably be an array of structs, which contain:
* node
* zone
* array of fragment counts

Should I change to that implementation, it should be relatively simple to implement.